llvm/include/llvm/Support/GenericDomTreeConstruction.h

```diff
@@ -197,10 +197,9 @@ struct SemiNCAInfo {
       constexpr bool Direction = IsReverse != IsPostDom;  // XOR.
       auto Successors = getChildren<Direction>(BB, BatchUpdates);
       if (SuccOrder && Successors.size() > 1)
-        llvm::sort(
-            Successors.begin(), Successors.end(), [=](NodePtr A, NodePtr B) {
-              return SuccOrder->find(A)->second < SuccOrder->find(B)->second;
-            });
+        llvm::sort(Successors, [SuccOrder](NodePtr A, NodePtr B) {
```

Comment:
use `[&]` capture? I don't think it's helpful to fuss about what's being captured for a locally referenced lambda like this.